[BUG] Invalid subnet CIDR block causes kube-ovn-cni pods to panic and enter crashloopbackoff

### Kube-OVN Version

v1.11.5

### Kubernetes Version

v1.28.6

### Operation-system/Kernel Version

"Ubuntu 22.04.5 LTS"

5.15.0-124-generic

### Description

A subnet created with an invalid CIDR causes the kube-ovn-cni pods to panic and enter a crashloopbackoff.

Instead, there should be validation on the API and the kubectl command to create the subnet with the inccorect cidr should fail, with a clear message explaining that the CIDR block was of the wrong format.

In addition, there should be validation in the kube-ovn code, so that if it tries to parse a CIDR block and it can't, it handles the error and does not panic.

Pod in crashloopbackoff yaml:
```
kubectl get subnet belle-test-3 -oyaml
apiVersion: kubeovn.io/v1
kind: Subnet
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"kubeovn.io/v1","kind":"Subnet","metadata":{"annotations":{},"name":"belle-test-3"},"spec":{"cidrBlock":"101/64"}}
  creationTimestamp: "2024-10-30T17:22:51Z"
  deletionGracePeriodSeconds: 0
  deletionTimestamp: "2024-10-31T13:45:03Z"
  finalizers:
  - kube-ovn-controller
  generation: 4
  name: belle-test-3
  resourceVersion: "157906490"
  uid: c2d755f2-f93a-469b-ad46-435ba74a5d52
spec:
  cidrBlock: 101/64
  default: false
  excludeIps:
  - 2002:0:0:1234::1
  gateway: 2002:0:0:1234::1
  gatewayNode: ""
  gatewayType: distributed
  natOutgoing: false
  private: false
  protocol: IPv6
  provider: ovn
  vpc: ovn-cluster
status:
  activateGateway: ""
  conditions:
  - lastTransitionTime: "2024-10-31T13:28:19Z"
    lastUpdateTime: "2024-10-31T13:28:19Z"
    message: subnet belle-test-3 cidr 2002:0:0:1234::/64 is conflict with subnet belle-test-2
      cidr 2002:0:0:1234::/64
    reason: ValidateLogicalSwitchFailed
    status: "False"
    type: Validated
  - lastTransitionTime: "2024-10-30T17:22:54Z"
    lastUpdateTime: "2024-10-30T17:22:54Z"
    message: subnet belle-test-3 cidr 2002:0:0:1234::/64 is conflict with subnet belle-test-2
      cidr 2002:0:0:1234::/64
    reason: ValidateLogicalSwitchFailed
    status: "True"
    type: Error
  - lastTransitionTime: "2024-10-30T17:22:54Z"
    lastUpdateTime: "2024-10-30T17:22:54Z"
    message: subnet belle-test-3 cidr 2002:0:0:1234::/64 is conflict with subnet belle-test-2
      cidr 2002:0:0:1234::/64
    reason: ValidateLogicalSwitchFailed
    status: "False"
    type: Ready
  dhcpV4OptionsUUID: ""
  dhcpV6OptionsUUID: ""
  u2oInterconnectionIP: ""
  v4availableIPs: 0
  v4usingIPs: 0
  v6availableIPs: 1.8446744073709552e+19
  v6usingIPs: 0
```
Logs on kube-ovn-cni pod:
```
E1101 16:04:53.669762 3360998 runtime.go:79] Observed a panic: &logrus.Entry{Logger:(*logrus.Logger)(0xc000264980), Data:logrus.Fields{"cidr":"101/64", "error":(*net.ParseError)(0xc000804440)}, Time:time.Date(2024, time.November, 1, 16, 4, 53, 669293613, time.Local), Level:0x0, Caller:(*runtime.Frame)(nil), Message:"Failed to parse CIDR", Buffer:(*bytes.Buffer)(nil), Context:context.Context(nil), err:""} (&{0xc000264980 map[cidr:101/64 error:invalid CIDR address: 101/64] 2024-11-01 16:04:53.669293613 +0000 UTC m=+0.516971737 panic <nil> Failed to parse CIDR <nil> <nil> })
goroutine 94 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x56a199a28240?, 0xc0003b5180})
        /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.26.0/pkg/util/runtime/runtime.go:75 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x0?})
        /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.26.0/pkg/util/runtime/runtime.go:49 +0x75
panic({0x56a199a28240, 0xc0003b5180})
        /opt/hostedtoolcache/go/1.19.9/x64/src/runtime/panic.go:884 +0x212
github.com/sirupsen/logrus.(*Entry).log(0xc0003b50a0, 0x0, {0xc000654018, 0x14})
        /home/runner/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:260 +0x4a7
github.com/sirupsen/logrus.(*Entry).Log(0xc0003b50a0, 0x0, {0xc000b7d6e8?, 0x4?, 0xc000804440?})
        /home/runner/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:304 +0x4f
github.com/sirupsen/logrus.(*Entry).Panic(...)
        /home/runner/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:342
github.com/alauda/felix/ip.MustParseCIDROrIP({0xc000a9e2a0, 0x6})
        /home/runner/go/pkg/mod/github.com/kubeovn/felix@v0.0.0-20220325073257-c8a0f705d139/ip/ip_addr.go:214 +0x177
github.com/alauda/felix/ipsets.IPSetType.CanonicaliseMember({0x56a1988d1aff?, 0xc000b01bc8?}, {0xc000a9e2a0, 0x6})
        /home/runner/go/pkg/mod/github.com/kubeovn/felix@v0.0.0-20220325073257-c8a0f705d139/ipsets/ipset_defs.go:172 +0x159
github.com/alauda/felix/ipsets.(*IPSets).filterAndCanonicaliseMembers(0xc000292480, {0x56a1988d1aff, 0x8}, {0xc0009426e0, 0x8, 0xc0000fc210?})
        /home/runner/go/pkg/mod/github.com/kubeovn/felix@v0.0.0-20220325073257-c8a0f705d139/ipsets/ipsets.go:238 +0xed
github.com/alauda/felix/ipsets.(*IPSets).AddOrReplaceIPSet(0xc000292480, {{0x56a1988d0646, 0x7}, {0x56a1988d1aff, 0x8}, 0x100000}, {0xc0009426e0, 0x8, 0xb})
        /home/runner/go/pkg/mod/github.com/kubeovn/felix@v0.0.0-20220325073257-c8a0f705d139/ipsets/ipsets.go:120 +0x22e
github.com/kubeovn/kube-ovn/pkg/daemon.(*Controller).setIPSet(0xc000a21600)
        /home/runner/work/kube-ovn/kube-ovn/pkg/daemon/gateway_linux.go:94 +0x31a
github.com/kubeovn/kube-ovn/pkg/daemon.(*Controller).Run(0xc000a21600, 0xc00051d1a0)
        /home/runner/work/kube-ovn/kube-ovn/pkg/daemon/controller.go:582 +0x3f7
created by github.com/kubeovn/kube-ovn/cmd/daemon.CmdMain
        /home/runner/work/kube-ovn/kube-ovn/cmd/daemon/cniserver.go:83 +0x5ca
panic: (*logrus.Entry) 0xc0003b5180 [recovered]
        panic: (*logrus.Entry) 0xc0003b5180

goroutine 94 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x0?})
        /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.26.0/pkg/util/runtime/runtime.go:56 +0xd7
panic({0x56a199a28240, 0xc0003b5180})
        /opt/hostedtoolcache/go/1.19.9/x64/src/runtime/panic.go:884 +0x212
github.com/sirupsen/logrus.(*Entry).log(0xc0003b50a0, 0x0, {0xc000654018, 0x14})
        /home/runner/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:260 +0x4a7
github.com/sirupsen/logrus.(*Entry).Log(0xc0003b50a0, 0x0, {0xc000b7d6e8?, 0x4?, 0xc000804440?})
        /home/runner/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:304 +0x4f
github.com/sirupsen/logrus.(*Entry).Panic(...)
        /home/runner/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:342
github.com/alauda/felix/ip.MustParseCIDROrIP({0xc000a9e2a0, 0x6})
        /home/runner/go/pkg/mod/github.com/kubeovn/felix@v0.0.0-20220325073257-c8a0f705d139/ip/ip_addr.go:214 +0x177
github.com/alauda/felix/ipsets.IPSetType.CanonicaliseMember({0x56a1988d1aff?, 0xc000b01bc8?}, {0xc000a9e2a0, 0x6})
        /home/runner/go/pkg/mod/github.com/kubeovn/felix@v0.0.0-20220325073257-c8a0f705d139/ipsets/ipset_defs.go:172 +0x159
github.com/alauda/felix/ipsets.(*IPSets).filterAndCanonicaliseMembers(0xc000292480, {0x56a1988d1aff, 0x8}, {0xc0009426e0, 0x8, 0xc0000fc210?})
        /home/runner/go/pkg/mod/github.com/kubeovn/felix@v0.0.0-20220325073257-c8a0f705d139/ipsets/ipsets.go:238 +0xed
github.com/alauda/felix/ipsets.(*IPSets).AddOrReplaceIPSet(0xc000292480, {{0x56a1988d0646, 0x7}, {0x56a1988d1aff, 0x8}, 0x100000}, {0xc0009426e0, 0x8, 0xb})
        /home/runner/go/pkg/mod/github.com/kubeovn/felix@v0.0.0-20220325073257-c8a0f705d139/ipsets/ipsets.go:120 +0x22e
github.com/kubeovn/kube-ovn/pkg/daemon.(*Controller).setIPSet(0xc000a21600)
        /home/runner/work/kube-ovn/kube-ovn/pkg/daemon/gateway_linux.go:94 +0x31a
github.com/kubeovn/kube-ovn/pkg/daemon.(*Controller).Run(0xc000a21600, 0xc00051d1a0)
        /home/runner/work/kube-ovn/kube-ovn/pkg/daemon/controller.go:582 +0x3f7
created by github.com/kubeovn/kube-ovn/cmd/daemon.CmdMain
        /home/runner/work/kube-ovn/kube-ovn/cmd/daemon/cniserver.go:83 +0x5ca
```

### Steps To Reproduce

1. Create file "subnet.yaml" with contents:
```
apiVersion: kubeovn.io/v1
kind: Subnet
metadata:
  name: broken-cidr-subnet
spec:
  cidrBlock: 104/64
```
2. Run `kubectl apply -f subnet.yaml`
3. Run `kubectl get pods -n kube-system` and see the kube-ovn-cni pods in crashloopbackackoff


### Current Behavior

Panics, and the kube-ovn-cni pod enters a crashloopbackoff.

Pod in crashloopbackoff yaml:
```
kubectl get subnet belle-test-3 -oyaml
apiVersion: kubeovn.io/v1
kind: Subnet
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"kubeovn.io/v1","kind":"Subnet","metadata":{"annotations":{},"name":"belle-test-3"},"spec":{"cidrBlock":"101/64"}}
  creationTimestamp: "2024-10-30T17:22:51Z"
  deletionGracePeriodSeconds: 0
  deletionTimestamp: "2024-10-31T13:45:03Z"
  finalizers:
  - kube-ovn-controller
  generation: 4
  name: belle-test-3
  resourceVersion: "157906490"
  uid: c2d755f2-f93a-469b-ad46-435ba74a5d52
spec:
  cidrBlock: 101/64
  default: false
  excludeIps:
  - 2002:0:0:1234::1
  gateway: 2002:0:0:1234::1
  gatewayNode: ""
  gatewayType: distributed
  natOutgoing: false
  private: false
  protocol: IPv6
  provider: ovn
  vpc: ovn-cluster
status:
  activateGateway: ""
  conditions:
  - lastTransitionTime: "2024-10-31T13:28:19Z"
    lastUpdateTime: "2024-10-31T13:28:19Z"
    message: subnet belle-test-3 cidr 2002:0:0:1234::/64 is conflict with subnet belle-test-2
      cidr 2002:0:0:1234::/64
    reason: ValidateLogicalSwitchFailed
    status: "False"
    type: Validated
  - lastTransitionTime: "2024-10-30T17:22:54Z"
    lastUpdateTime: "2024-10-30T17:22:54Z"
    message: subnet belle-test-3 cidr 2002:0:0:1234::/64 is conflict with subnet belle-test-2
      cidr 2002:0:0:1234::/64
    reason: ValidateLogicalSwitchFailed
    status: "True"
    type: Error
  - lastTransitionTime: "2024-10-30T17:22:54Z"
    lastUpdateTime: "2024-10-30T17:22:54Z"
    message: subnet belle-test-3 cidr 2002:0:0:1234::/64 is conflict with subnet belle-test-2
      cidr 2002:0:0:1234::/64
    reason: ValidateLogicalSwitchFailed
    status: "False"
    type: Ready
  dhcpV4OptionsUUID: ""
  dhcpV6OptionsUUID: ""
  u2oInterconnectionIP: ""
  v4availableIPs: 0
  v4usingIPs: 0
  v6availableIPs: 1.8446744073709552e+19
  v6usingIPs: 0
```
Logs on kube-ovn-cni pod:
```
E1101 16:04:53.669762 3360998 runtime.go:79] Observed a panic: &logrus.Entry{Logger:(*logrus.Logger)(0xc000264980), Data:logrus.Fields{"cidr":"101/64", "error":(*net.ParseError)(0xc000804440)}, Time:time.Date(2024, time.November, 1, 16, 4, 53, 669293613, time.Local), Level:0x0, Caller:(*runtime.Frame)(nil), Message:"Failed to parse CIDR", Buffer:(*bytes.Buffer)(nil), Context:context.Context(nil), err:""} (&{0xc000264980 map[cidr:101/64 error:invalid CIDR address: 101/64] 2024-11-01 16:04:53.669293613 +0000 UTC m=+0.516971737 panic <nil> Failed to parse CIDR <nil> <nil> })
goroutine 94 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x56a199a28240?, 0xc0003b5180})
        /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.26.0/pkg/util/runtime/runtime.go:75 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x0?})
        /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.26.0/pkg/util/runtime/runtime.go:49 +0x75
panic({0x56a199a28240, 0xc0003b5180})
        /opt/hostedtoolcache/go/1.19.9/x64/src/runtime/panic.go:884 +0x212
github.com/sirupsen/logrus.(*Entry).log(0xc0003b50a0, 0x0, {0xc000654018, 0x14})
        /home/runner/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:260 +0x4a7
github.com/sirupsen/logrus.(*Entry).Log(0xc0003b50a0, 0x0, {0xc000b7d6e8?, 0x4?, 0xc000804440?})
        /home/runner/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:304 +0x4f
github.com/sirupsen/logrus.(*Entry).Panic(...)
        /home/runner/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:342
github.com/alauda/felix/ip.MustParseCIDROrIP({0xc000a9e2a0, 0x6})
        /home/runner/go/pkg/mod/github.com/kubeovn/felix@v0.0.0-20220325073257-c8a0f705d139/ip/ip_addr.go:214 +0x177
github.com/alauda/felix/ipsets.IPSetType.CanonicaliseMember({0x56a1988d1aff?, 0xc000b01bc8?}, {0xc000a9e2a0, 0x6})
        /home/runner/go/pkg/mod/github.com/kubeovn/felix@v0.0.0-20220325073257-c8a0f705d139/ipsets/ipset_defs.go:172 +0x159
github.com/alauda/felix/ipsets.(*IPSets).filterAndCanonicaliseMembers(0xc000292480, {0x56a1988d1aff, 0x8}, {0xc0009426e0, 0x8, 0xc0000fc210?})
        /home/runner/go/pkg/mod/github.com/kubeovn/felix@v0.0.0-20220325073257-c8a0f705d139/ipsets/ipsets.go:238 +0xed
github.com/alauda/felix/ipsets.(*IPSets).AddOrReplaceIPSet(0xc000292480, {{0x56a1988d0646, 0x7}, {0x56a1988d1aff, 0x8}, 0x100000}, {0xc0009426e0, 0x8, 0xb})
        /home/runner/go/pkg/mod/github.com/kubeovn/felix@v0.0.0-20220325073257-c8a0f705d139/ipsets/ipsets.go:120 +0x22e
github.com/kubeovn/kube-ovn/pkg/daemon.(*Controller).setIPSet(0xc000a21600)
        /home/runner/work/kube-ovn/kube-ovn/pkg/daemon/gateway_linux.go:94 +0x31a
github.com/kubeovn/kube-ovn/pkg/daemon.(*Controller).Run(0xc000a21600, 0xc00051d1a0)
        /home/runner/work/kube-ovn/kube-ovn/pkg/daemon/controller.go:582 +0x3f7
created by github.com/kubeovn/kube-ovn/cmd/daemon.CmdMain
        /home/runner/work/kube-ovn/kube-ovn/cmd/daemon/cniserver.go:83 +0x5ca
panic: (*logrus.Entry) 0xc0003b5180 [recovered]
        panic: (*logrus.Entry) 0xc0003b5180

goroutine 94 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x0?})
        /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.26.0/pkg/util/runtime/runtime.go:56 +0xd7
panic({0x56a199a28240, 0xc0003b5180})
        /opt/hostedtoolcache/go/1.19.9/x64/src/runtime/panic.go:884 +0x212
github.com/sirupsen/logrus.(*Entry).log(0xc0003b50a0, 0x0, {0xc000654018, 0x14})
        /home/runner/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:260 +0x4a7
github.com/sirupsen/logrus.(*Entry).Log(0xc0003b50a0, 0x0, {0xc000b7d6e8?, 0x4?, 0xc000804440?})
        /home/runner/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:304 +0x4f
github.com/sirupsen/logrus.(*Entry).Panic(...)
        /home/runner/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:342
github.com/alauda/felix/ip.MustParseCIDROrIP({0xc000a9e2a0, 0x6})
        /home/runner/go/pkg/mod/github.com/kubeovn/felix@v0.0.0-20220325073257-c8a0f705d139/ip/ip_addr.go:214 +0x177
github.com/alauda/felix/ipsets.IPSetType.CanonicaliseMember({0x56a1988d1aff?, 0xc000b01bc8?}, {0xc000a9e2a0, 0x6})
        /home/runner/go/pkg/mod/github.com/kubeovn/felix@v0.0.0-20220325073257-c8a0f705d139/ipsets/ipset_defs.go:172 +0x159
github.com/alauda/felix/ipsets.(*IPSets).filterAndCanonicaliseMembers(0xc000292480, {0x56a1988d1aff, 0x8}, {0xc0009426e0, 0x8, 0xc0000fc210?})
        /home/runner/go/pkg/mod/github.com/kubeovn/felix@v0.0.0-20220325073257-c8a0f705d139/ipsets/ipsets.go:238 +0xed
github.com/alauda/felix/ipsets.(*IPSets).AddOrReplaceIPSet(0xc000292480, {{0x56a1988d0646, 0x7}, {0x56a1988d1aff, 0x8}, 0x100000}, {0xc0009426e0, 0x8, 0xb})
        /home/runner/go/pkg/mod/github.com/kubeovn/felix@v0.0.0-20220325073257-c8a0f705d139/ipsets/ipsets.go:120 +0x22e
github.com/kubeovn/kube-ovn/pkg/daemon.(*Controller).setIPSet(0xc000a21600)
        /home/runner/work/kube-ovn/kube-ovn/pkg/daemon/gateway_linux.go:94 +0x31a
github.com/kubeovn/kube-ovn/pkg/daemon.(*Controller).Run(0xc000a21600, 0xc00051d1a0)
        /home/runner/work/kube-ovn/kube-ovn/pkg/daemon/controller.go:582 +0x3f7
created by github.com/kubeovn/kube-ovn/cmd/daemon.CmdMain
        /home/runner/work/kube-ovn/kube-ovn/cmd/daemon/cniserver.go:83 +0x5ca
```

### Expected Behavior

Kubectl command to apply the subnet with the incorrect CIDR fails with clear message as to why e.g. "Invalid CIDR Block: should be of form ..."

--- a/pkg/daemon/gateway.go
+++ b/pkg/daemon/gateway.go
@@ -100,8 +100,10 @@ func (c *Controller) getSubnetsNeedNAT(protocol string) ([]string, error) {
 
 	for _, subnet := range subnets {
 		if c.isSubnetNeedNat(subnet, protocol) {
-			cidrBlock := getCidrByProtocol(subnet.Spec.CIDRBlock, protocol)
-			subnetsNeedNat = append(subnetsNeedNat, cidrBlock)
+			cidrBlock, err := getCidrByProtocol(subnet.Spec.CIDRBlock, protocol)
+			if err == nil {
+				subnetsNeedNat = append(subnetsNeedNat, cidrBlock)
+			}
 		}
 	}
 	return subnetsNeedNat, nil
@@ -138,8 +140,10 @@ func (c *Controller) getSubnetsDistributedGateway(protocol string) ([]string, er
 			subnet.Spec.CIDRBlock != "" &&
 			subnet.Spec.GatewayType == kubeovnv1.GWDistributedType &&
 			(subnet.Spec.Protocol == kubeovnv1.ProtocolDual || subnet.Spec.Protocol == protocol) {
-			cidrBlock := getCidrByProtocol(subnet.Spec.CIDRBlock, protocol)
-			result = append(result, cidrBlock)
+			cidrBlock, err := getCidrByProtocol(subnet.Spec.CIDRBlock, protocol)
+			if err == nil {
+				result = append(result, cidrBlock)
+			}
 		}
 	}
 	return result, nil
@@ -167,9 +171,11 @@ func (c *Controller) getDefaultVpcSubnetsCIDR(protocol string) ([]string, map[st
 
 	for _, subnet := range subnets {
 		if subnet.Spec.Vpc == c.config.ClusterRouter && (subnet.Spec.Vlan == "" || subnet.Spec.LogicalGateway) && subnet.Spec.CIDRBlock != "" {
-			cidrBlock := getCidrByProtocol(subnet.Spec.CIDRBlock, protocol)
-			ret = append(ret, cidrBlock)
-			subnetMap[subnet.Name] = cidrBlock
+			cidrBlock, err := getCidrByProtocol(subnet.Spec.CIDRBlock, protocol)
+			if err == nil {
+				ret = append(ret, cidrBlock)
+				subnetMap[subnet.Name] = cidrBlock
+			}
 		}
 	}
 	return ret, subnetMap, nil
@@ -197,8 +203,12 @@ func (c *Controller) getOtherNodes(protocol string) ([]string, error) {
 	return ret, nil
 }
 
-func getCidrByProtocol(cidr, protocol string) string {
+func getCidrByProtocol(cidr, protocol string) (string, error) {
 	var cidrStr string
+	if err := util.CheckCidrs(cidr); err != nil {
+		return "", err
+	}
+
 	if util.CheckProtocol(cidr) == kubeovnv1.ProtocolDual {
 		cidrBlocks := strings.Split(cidr, ",")
 		if protocol == kubeovnv1.ProtocolIPv4 {
@@ -209,7 +219,7 @@ func getCidrByProtocol(cidr, protocol string) string {
 	} else {
 		cidrStr = cidr
 	}
-	return cidrStr
+	return cidrStr, nil
 }
 
 func (c *Controller) getEgressNatIPByNode(nodeName string) (map[string]string, error) {

--- a/pkg/daemon/gateway_linux.go
+++ b/pkg/daemon/gateway_linux.go
@@ -208,7 +208,11 @@ func (c *Controller) reconcileNatOutGoingPolicyIPset(protocol string) {
 	subnetCidrs := make([]string, 0)
 	natPolicyRuleIDs := strset.New()
 	for _, subnet := range subnets {
-		cidrBlock := getCidrByProtocol(subnet.Spec.CIDRBlock, protocol)
+		cidrBlock, err := getCidrByProtocol(subnet.Spec.CIDRBlock, protocol)
+		if err != nil {
+			klog.Errorf("failed to get subnet %s CIDR block by protocol: %v", subnet.Name, err)
+			continue
+		}
 		subnetCidrs = append(subnetCidrs, cidrBlock)
 		for _, rule := range subnet.Status.NatOutgoingPolicyRules {
 			if rule.RuleID == "" {
@@ -994,7 +998,11 @@ func (c *Controller) generateNatOutgoingPolicyChainRules(protocol string) ([]uti
 		subnet := subnetMap[subnetName]
 		var natPolicyRuleIptables []util.IPTableRule
 		natPolicySubnetUIDs.Add(util.GetTruncatedUID(string(subnet.GetUID())))
-		cidrBlock := getCidrByProtocol(subnet.Spec.CIDRBlock, protocol)
+		cidrBlock, err := getCidrByProtocol(subnet.Spec.CIDRBlock, protocol)
+		if err != nil {
+			klog.Errorf("failed to get subnet %s cidr block with protocol: %v", subnet.Name, err)
+			continue
+		}
 
 		ovnNatPolicySubnetChainName := OvnNatOutGoingPolicySubnet + util.GetTruncatedUID(string(subnet.GetUID()))
 		natPolicySubnetIptables = append(natPolicySubnetIptables, util.IPTableRule{Table: NAT, Chain: OvnNatOutGoingPolicy, Rule: strings.Fields(fmt.Sprintf(`-s %s -m comment --comment natPolicySubnet-%s -j %s`, cidrBlock, subnet.Name, ovnNatPolicySubnetChainName))})
@@ -1607,8 +1615,10 @@ func (c *Controller) getSubnetsNeedPR(protocol string) (map[policyRouteMeta]stri
 				meta.gateway = egw[0]
 			}
 			if meta.gateway != "" {
-				cidrBlock := getCidrByProtocol(subnet.Spec.CIDRBlock, protocol)
-				subnetsNeedPR[meta] = cidrBlock
+				cidrBlock, err := getCidrByProtocol(subnet.Spec.CIDRBlock, protocol)
+				if err == nil {
+					subnetsNeedPR[meta] = cidrBlock
+				}
 			}
 		}
 	}

